### PR TITLE
[core][ci] Add some core release tests to autoscaler v2 project for nightly tests

### PR DESCRIPTION
**Pull request by rickyyx**
## Why are these changes needed?
This adds a variant of release tests for core that would use Autoscaler V2 when run. 

**What's changed**
This means for the tests changed in this PR, they will run on both V1 and V2 autoscaler. 

**Which tests**
Tests that are:
1. core nightly tests AND
2. use multiple nodes

i.e.: 
- stress_test_many_tasks
- stress_test_dead_actors
- threaded_actors_stress_test
- stress_test_many_runtime_envs
- many_nodes_actor_test_on_v2
- pg_autoscaling_regression_test
- placement_group_performance_test
- object_store
- many_actors
- many_tasks
- many_pgs
- many_nodes
- scheduling_test_many_0s_tasks_many_nodes
- 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/ray_release/environments/aws_pa.env [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# For Integration with Anyscale Autoscaler
+ANYSCALE_HOST=https://console.anyscale-staging.com
+RELEASE_AWS_ANYSCALE_SECRET_ARN="arn:aws:secretsmanager:us-west-2:029272617770:secret:release-automation/anyscale-staging-token20221014164754935800000001-pfQunc"
+RELEASE_DEFAULT_CLOUD_ID="cld_kvedZWag2qA8i5BjxUevf5i7"
+RELEASE_DEFAULT_PROJECT="prj_a1x9vj2wl72sl67v5myw6f5en3"
+ANYSCALE_PROJECT="prj_a1x9vj2wl72sl67v5myw6f5en3"
```
Comment (aslonnie) [inline release/ray_release/environments/aws_pa.env:6]: nit: eof eol?

File: release/release_tests.yaml [hunk 2]
```diff
@@ -5107,6 +5107,8 @@
         cluster_compute: stress_tests/stress_tests_compute_gce.yaml
       smoke_test:
         frequency: manual
+    - __suffix__: aws_pa
+      env: aws_pa
```
Comment (aslonnie) [inline release/release_tests.yaml:5111]: what does the `aws_pa` thing mean?

Comment (rickyyx) [reply release/release_tests.yaml:5111]: pa: product autoscaler

Comment (aslonnie) [reply release/release_tests.yaml:5111]: I am not sure if this should be another "env", or maybe should be something clearer, but will leave this to @can-anyscale to judge. :)

Comment (rickyyx) [reply release/release_tests.yaml:5111]: i guess we just need another product project - which is coded as part of the env. 

This is also more of an intermediate state - we will eventally move everything to "aws_pa" (aws -> aws_pa)

**Review by scv119 — APPROVED** [review]

**Review by rkooo567 — APPROVED** [review]

**Comment by rickyyx** [comment]
Staging already running with PA.

**Pull request closed without merging**